### DISCUSSION TITLE
Is it possible to change the key that is used during Cell/Point Selection?

### DISCUSSION THREAD

**Discussion by JacobBumgarner** [pyvista/pyvista#2513] [Q&A]
I've used PyVista and PyVistaQt in a vasculature analysis application I've built, [VesselVio](https://github.com/JacobBumgarner/VesselVio). Now, I'm hoping to add a feature where users can select individual vessels in order to be able to identify specific features of interest.

I've created a prototyped solution using `plotter.enable_cell_picking()`, where the user presses 'P' to select vessels underneath the pointer. A temporary vessel mesh is then added to to this location to show that the vessel has been 'selected'.

**Questions:**
1. How can I remove/disable the 'R' key observer from the `plotter.enable_cell_picking()`? I'd like users to still be able to reset the plotter view when picking is enabled.
2. Somewhat related, is there any way to change the observer key 'P' to a different key?

I've dug into the `PickingHelper.enable_cell_picking()` a bit, but I haven't been able to figure it out.

<details>
<summary>Dropdown for the code!</summary>

```python
import numpy as np
import pyvista as pv

p = pv.Plotter()

tubes = pv.MultiBlock()
tube_info = {}
for i in range(100):
    # Build coordinates for a random tube
    theta = np.linspace(0, np.random.randint(1, 8)*np.pi, 100)
    x = np.cos(theta)
    y = np.sin(theta)
    z = np.linspace(0, np.random.randint(1, 30), 100)
    coords = np.stack([x,y,z], axis=1)
    coords += np.random.randint(-50, 50, 3) # Translate the tube
    r = np.random.rand(3, 3) * 2 * np.pi # Random rotation
    rot_matrix = np.array([[np.cos(r[0][0]), -np.sin(r[0][1]), np.sin(r[0][2])],
                           [np.sin(r[1][0]), np.cos(r[1][1]), -np.sin(r[1][2])],
                           [-np.sin(r[2][0]), np.sin(r[2][1]), np.cos(r[2][2])]])
    coords = np.dot(coords, rot_matrix) # Rotate the tube
    
    # Create the mesh
    spline = pv.Spline(coords)
    radius = np.random.rand(1)[0] * 3 # Random tube radius
    spline['vessel_id'] = [i] # vessel_ID scalar to identify the parent tube
    spline['radius'] = [radius]
    tube = spline.tube(radius=radius)
    
    tube_info[i] = {'coords': coords, 'radius': radius} # Add to tube dict
    tubes.append(tube) # Add to the multiblock list
    
block = tubes.combine() # Combine the multiblock

p.add_mesh(block)

def tube_identifier(grid):
    """A function to identify the id of the selected mesh"""
    if not grid:
        p.remove_actor(p._picked_vessel)
        p._picked_vessel = None
        return
    v_id = grid.cell_data['vessel_id']
    tube_values = tube_info[v_id[0]]
    spline = pv.Spline(tube_values['coords'])
    tube = spline.tube(tube_values['radius']+0.05)
    
    if p._picked_vessel:
        p.remove_actor(p._picked_vessel)
    p._picked_vessel = p.add_mesh(tube, color='red')
    print(f"Tube ID: {v_id[0]}. Tube Radius: {tube_values['radius']}")

p._picked_vessel = None # A shady temporary private attribute for the demo

p.enable_cell_picking(callback=tube_identifier, through=False,
                      show=False)
p.show()

```

</details>

Here is a video demo as well.

https://user-images.githubusercontent.com/70919881/165176918-1ec08be8-45a8-444f-8104-861b6ca1da96.mov

**Comment by banesullivan** [comment]
> Somewhat related, is there any way to change the observer key 'P' to a different key?

We recently added support for clicking with point-picking. We could refactor all of this to allow the user to specify a custom key or mouse button

**Comment by banesullivan** [comment]
You may want to implement your own variant of `enable_cell_picking()` that uses a different interaction style. the cell picking uses`enable_rubber_band_style()` which comes with that `R`-> rubber band functionality
